src/main/java/org/broadinstitute/hellbender/tools/exome/germlinehmm/IntegerCopyNumberTransitionMatrixCollection.java

```diff
@@ -0,0 +1,167 @@
+                .collect(Collectors.toSet());
+    }
+
+    /**
+     * Returns a set of contigs that are annotated at least once
+     *
+     * @return
+     */
+    public Set<String> getContigs() {
+        return sexGenotypeAndContigToTransitionMatrixMap.keySet().stream()
+                .map(SexGenotypeContigPairKey::getContig)
+                .collect(Collectors.toSet());
+    }
+
+    public static IntegerCopyNumberTransitionMatrixCollection read(final File tableFile) {
+        return IntegerCopyNumberTransitionMatrixCollectionReader.read(tableFile);
+    }
+
+    public static IntegerCopyNumberTransitionMatrixCollection read(final Reader reader,
+                                                                   final String parentPath) {
+        return IntegerCopyNumberTransitionMatrixCollectionReader.read(reader, parentPath);
+    }
+
+    /**
+     * Asserts that all of provided sex genotypes and contigs have assigned copy number transition
+     * matrices.
+     *
+     * @param sexGenotypes a set of sex genotype identifier strings
+     * @param contigs      a set of contig strings
+     * @throws UserException.BadInput if a required data entry is not available
+     */
+    public void assertCompleteness(@Nonnull final Set<String> sexGenotypes,
+                                   @Nonnull final Set<String> contigs) {
+        for (final String sexGenotype : sexGenotypes) {
+            for (final String contig : contigs) {
+                checkKey(sexGenotype, contig);
+            }
+        }
+    }
+
+    /**
+     * Asserts that all pairwise choices of (complete sex genotype, contig) are annotated
+     */
+    public void assertCompleteness() {
+        assertCompleteness(getSexGenotypes(), getContigs());
+    }
+
+    public int getMaxCopyNumberInCollection() {
+        return maxCopyNumberInCollection;
+    }
+
+    /**
+     *
+     * @param sexGenotype
+     * @param contig
+     * @return
+     */
+    private SexGenotypeContigPairKey checkKey(@Nonnull final String sexGenotype,
+                                              @Nonnull final String contig) {
+        final SexGenotypeContigPairKey key = SexGenotypeContigPairKey.of(sexGenotype, contig);
+        if (!sexGenotypeAndContigToTransitionMatrixMap.containsKey(key)) {
+            throw new UserException.BadInput(String.format("The collection does not include" +
+                            " the integer copy number transition matrix for sex genotype \"%s\" and contig \"%s\"",
+                    sexGenotype, contig));
+        }
+        return key;
+    }
+
+    /**
+     * Finds the highest copy number for all sex genotypes and contigs, returns a clone with all transition
+     * matrices padded to the highest copy number.
+     *
+     * @return a new instance of {@link IntegerCopyNumberTransitionMatrixCollection}
+     */
+    public IntegerCopyNumberTransitionMatrixCollection cloneWithPaddedStates() {
```

Comment:
In fact, I thought (am thinking) about allowing HMMs to have a different set of hidden states on different targets. Polymorphic regions and those known for STR can have very high CN states but it shouldn't come at a global quadratic complexity cost.